perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/AppAutoRejectionScheduler.java

```diff
@@ -0,0 +1,252 @@
+					String reasonForGroup = application.getGroup() == null ? "" : "Your application to group " + application.getGroup().getName() + " was auto rejected, because you didn't verify your email address.";
+					rejectWithReason(application, reasonForVo, reasonForGroup);
+					continue;
+				}
+			}
+			int expirationAppIgnoredByAdmin = Integer.parseInt(attrValue.get("ignoredByAdmin"));
+			if (now.minusDays(expirationAppIgnoredByAdmin).isAfter(modifiedAt)) {
+				String reasonForVo = "Your application to VO " + application.getVo().getName() + " was auto rejected, because admin didn't approve your application in a timely manner.";
+				String reasonForGroup = application.getGroup() == null ? "" : "Your application to group " + application.getGroup().getName() + " was auto rejected, because admin didn't approve your application in a timely manner.";
+				rejectWithReason(application, reasonForVo, reasonForGroup);
+			}
+		}
+	}
+
+	/**
+	 * Rejects given application to Vo or group due to given reason.
+	 *
+	 * @param application application to reject
+	 * @param reasonForVo reason for reject VO application
+	 * @param reasonForGroup reason for reject group application
+	 */
+	private void rejectWithReason (Application application, String reasonForVo, String reasonForGroup) {
+		try {
+			if (application.getGroup() == null) {
+				registrarManager.rejectApplication(sess, application.getId(), reasonForVo);
+			} else {
+				registrarManager.rejectApplication(sess, application.getId(), reasonForGroup);
+			}
+		} catch (PerunException e) {
+			log.error("Failed to reject expired application: {}", application, e);
+		}
+
+	}
+
+	/**
+	 * Selects all vos from database, in which could be some expired applications acceptable for auto rejection.
+	 *
+	 * @return list of vos
+	 * @throws VoNotExistsException if vo not exist (it shouldn't happen)
+	 */
+	private List<Vo> getAllEligibleVos() throws VoNotExistsException {
+		String sql = "select distinct vos.id from vos join application on vos.id=application.vo_id join vo_attr_values " +
+			"on vos.id=vo_attr_values.vo_id join attr_names on attr_names.id=vo_attr_values.attr_id " +
+			"where application.state in ('NEW', 'VERIFIED') and (attr_names.friendly_name= 'applicationExpirationRules' " +
+			"and attr_names.namespace= 'urn:perun:vo:attribute-def:def')";
+		List<Integer> vosIds = jdbc.queryForList(sql, Integer.class);
+		List<Vo> vos = new ArrayList<>();
+		for(int voId : vosIds){
+			vos.add(perun.getVosManagerBl().getVoById(sess, voId));
+		}
+		return vos;
+
+	}
+
+	/**
+	 * Selects all groups from database, in which could be some expired applications acceptable for auto rejection.
+	 *
+	 * @return list of groups
+	 * @throws GroupNotExistsException if group not exist (it shouldn't happen)
+	 */
+	private List<Group> getAllEligibleGroups() throws GroupNotExistsException {
```

Comment:
Same for this one.

Reply:
This query also moved to the SearcherImpl.